### PR TITLE
Halit-week2

### PR DESCRIPTION
**Pull request by halitos**
Tasks complete, 
API works fine on my local machine, server.js file attached.

### CODE REVIEW

File: week-2/mandatory/2-ecommerce-db/task.md [hunk 1]
```diff
@@ -7,8 +7,79 @@ In this homework, you are going to work with an ecommerce database. In this data
 Below you will find a set of tasks for you to complete to set up a database for an e-commerce app.
 
 To submit this homework write the correct commands for each question here:
+
 ```sql
 
+1. SELECT name, address FROM customers WHERE country='United States';
+
+2. SELECT * FROM customers ORDER BY name ASC;  -- it already orders asc by default --
```
Comment (Ashaghel) [inline week-2/mandatory/2-ecommerce-db/task.md:15]: like the comment! the default is for sure asc so no need to add !

File: week-2/mandatory/2-ecommerce-db/task.md [hunk 2]
```diff
@@ -7,8 +7,79 @@ In this homework, you are going to work with an ecommerce database. In this data
...
+
+3. SELECT * FROM products WHERE product_name LIKE '%socks%';
+
+4. SELECT products.id, products.product_name, pa.unit_price, pa.supp_id
+    FROM products JOIN product_availability pa ON products.id=pa.prod_id
+    WHERE pa.unit_price > 100;
```
Comment (Ashaghel) [inline week-2/mandatory/2-ecommerce-db/task.md:21]: nice work, best practice is to follow one way in giving aliases for tables, so you give products an alias as well

File: week-2/mandatory/2-ecommerce-db/task.md [hunk 3]
```diff
@@ -7,8 +7,79 @@ In this homework, you are going to work with an ecommerce database. In this data
...
+
+6. SELECT products.product_name, pa.unit_price, sup.supplier_name FROM
+    products JOIN product_availability pa ON
+    (products.id=pa.prod_id) JOIN suppliers sup ON
+    (pa.supp_id=sup.id);
+
+7. SELECT p.product_name, sup.supplier_name FROM
+    products p JOIN product_availability pa ON
+    (p.id=pa.prod_id) JOIN suppliers sup ON
+    (pa.supp_id=sup.id) WHERE sup.country='United Kingdom';
```
Comment (Ashaghel) [inline week-2/mandatory/2-ecommerce-db/task.md:36]: Nice work!

File: week-2/mandatory/2-ecommerce-db/task.md [hunk 4]
```diff
@@ -7,8 +7,79 @@ In this homework, you are going to work with an ecommerce database. In this data
...
+
+8. SELECT cust.id cust_id, orders.id order_id, orders.order_reference,
+    orders.order_date, oi.quantity * pa.unit_price as total_cost
+    FROM customers cust JOIN orders ON (cust.id=orders.customer_id)
+    JOIN order_items oi ON (orders.id=oi.order_id)
+    JOIN product_availability pa ON (oi.supplier_id=pa.supp_id)
+    WHERE cust.id=1;
```
Comment (Ashaghel) [inline week-2/mandatory/2-ecommerce-db/task.md:43]: The question is a bit not clear here, but it was meant to ask for orders and their total price (so grouping and summarising the order items for each order)
so if order A has 5 items, the total will be the sum of (quantity of each order * unit price)

File: week-2/mandatory/2-ecommerce-db/task.md [hunk 5]
```diff
@@ -7,8 +7,79 @@ In this homework, you are going to work with an ecommerce database. In this data
...
+
+9.
+    SELECT cust.id cust_id, cust.name cust_name, orders.id order_id, orders.order_date,
+    orders.order_reference, oi.product_id, oi.supplier_id, oi.quantity
+    FROM customers cust JOIN orders ON (cust.id=orders.customer_id)
+    JOIN order_items oi ON (orders.id=oi.order_id)
+    WHERE cust.name='Hope Crosby';
+
+ -- if i select orders.* and oi.* list duplicate unnecessary id numbers and hard to read, so tailored by selecting usefull fields..stingy customer btw--
```
Comment (Ashaghel) [inline week-2/mandatory/2-ecommerce-db/task.md:52]: hhhhhh nicely done, the question had its issues I think again.

File: week-2/mandatory/2-ecommerce-db/task.md [hunk 6]
```diff
@@ -7,8 +7,79 @@ In this homework, you are going to work with an ecommerce database. In this data
...
+    JOIN order_items oi ON (orders.id=oi.order_id)
+    JOIN products pro ON (oi.product_id=pro.id)
+    JOIN suppliers sup ON (oi.supplier_id=sup.id);
+
+12.
+    SELECT DISTINCT cust.name suspected_customers
+    FROM customers cust JOIN orders ON (cust.id=orders.customer_id)
+    JOIN order_items oi ON (orders.id=oi.order_id)
+    JOIN suppliers sup ON (oi.supplier_id=sup.id)
+    WHERE sup.country='China';
```
Comment (Ashaghel) [inline week-2/mandatory/2-ecommerce-db/task.md:74]: nice work on distinct

File: week-2/mandatory/2-ecommerce-db/task.md [hunk 7]
```diff
@@ -7,8 +7,79 @@ In this homework, you are going to work with an ecommerce database. In this data
...
+
+13.
+    SELECT cust.name cust_names, orders.order_reference, orders.order_date,
+    oi.quantity * pa.unit_price total_amount
+    FROM customers cust JOIN orders ON (cust.id=orders.customer_id)
+    JOIN order_items oi ON (orders.id=oi.order_id)
+    JOIN product_availability pa ON (oi.product_id=pa.prod_id)
+    ORDER BY total_amount DESC;
```
Comment (Ashaghel) [inline week-2/mandatory/2-ecommerce-db/task.md:84]: missing grouping here, as it looks for aggregating the value of total

File: week-2/mandatory/3-api/server.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+app.get("/", (req, res) => {
+  res.send("Hello Wellcome to CYF ecommerce App ");
+});
+
+app.get("/customers", function (req, res) {
+  db.query(`SELECT * FROM customers`, (error, result) => {
+    if (error) throw error;
+    res.json(result.rows);
+  });
+});
```
Comment (Ashaghel) [inline week-2/mandatory/3-api/server.js:28]: you're throwing the error but better to catch it somewhere (then and catch)

Comment (Ashaghel) [reply week-2/mandatory/3-api/server.js:28]: sending the user a better message (data unavailable ie)

**Review by Ashaghel — COMMENTED** [review]
Speedy and excellent!
only some small issues

File: week-2/mandatory/3-api/server.js (hunk 8 above)
Comment (halitos) [reply week-2/mandatory/3-api/server.js:28]: working on catching errors

File: week-2/mandatory/2-ecommerce-db/task.md (hunk 7 above)
Comment (halitos) [reply week-2/mandatory/2-ecommerce-db/task.md:84]: Groups added

**Comment by halitos** [comment]
Thank you again

**Pull request closed without merging**